README.md
This addon shows a popup next up notification automatically to prompt for playing the next unwatched episode.

Configuration:

  - The addon is found in the services section and allows the notification time to be adjusted (default 30 seconds before the end)
  - The default action (ie Play/Do not Play) when nothing is pressed can also be configured
  - The number of episodes played in a row with no intervention to bring up the still watching window can be adjusted.
 
Skinners:

  - There is a script-nextup-notification-NextUpInfo.xml and script-nextup-notification-StillWatchingInfo.xml file located in default/720p/ simply copy this to your skin folder and adjust it how you like it. 
  
      Controls Available:
  
      - script-nextup-notification-NextUpInfo.xml
          - 3000 - Title
          - 3001 - Plot
          - 3002 - Season/Episode
          - 3003 - Rating
          - 3004 - First Aired 
          - 3006 - TV Show ClearArt
          - 3008 - Episode Thumb
          - 3009 - TV Show Poster
          - 3010 - TV Show Landscape
          - 3012 - Watch Now Button
          - 3013 - Cancel Button
          
      - script-nextup-notification-StillWatchingInfo.xml
          - 4000 - Label
          - 4001 - TV Show Poster
          - 4002 - Episode Thumb
          - 4003 - Rating
          - 4004 - First Aired
          - 4005 - TV Show Landscape
          - 4006 - Plot
          - 4012 - Continue Watching Button
          - 4013 - Cancel Button

changelog.txt
[B]1.0.4[/B]
- Added Season Number, Episode Number, Video Resolution and TV Show Fanart for skin support

[B]1.0.3[/B]
- Does not show when pseudotvlive is active

[B]1.0.2[/B]
- Bug fix for playing same last episode
- Landscape image added for next up

[B]1.0.1[/B]
- Added Still Watching Info
- Minor layout adjustment NextUpInfo

[B]1.0.0[/B]
- Ready for release

[B]0.0.7[/B]
- bug fix

[B]0.0.6[/B]
- bug fix

[B]0.0.5[/B]
- rename nextup info.xml

[B]0.0.4[/B]
- Option to set play mode

[B]0.0.3[/B]
- Added episode image

[B]0.0.2[/B]
- Bug fixes.

[B]0.0.1[/B]
- Initial version

resources/lib/NextUpInfo.py

import sys
import xbmc
import xbmcgui
import xbmcaddon
import json as json
import urllib

ACTION_PLAYER_STOP = 13
    
class NextUpInfo(xbmcgui.WindowXMLDialog):

    item = None
    cancel = False
    watchnow = False

    def __init__(self, *args, **kwargs):
        xbmcgui.WindowXMLDialog.__init__(self, *args, **kwargs)

    def onInit(self):
        self.action_exitkeys_id = [10, 13]
    
        image = self.item['art'].get('tvshow.poster','')
        thumb = self.item['art'].get('thumb','')
        clearartimage = self.item['art'].get('tvshow.clearart','')
        landscapeimage = self.item['art'].get('tvshow.landscape','')
        fanartimage = self.item['art'].get('tvshow.fanart','')
        overview = self.item['plot']
        name = self.item['title']
        
        episodeInfo = ""
        season = self.item['season']
        episodeNum = self.item['episode']
        episodeInfo = str(season) + "x" + str(episodeNum) + "."
        
        rating = str(round(float(self.item['rating'])))
        year = self.item['firstaired']
        info = year
        # set the dialog data
        self.getControl(3000).setLabel(name)
        self.getControl(3001).setText(overview)
        self.getControl(3002).setLabel(episodeInfo)
        self.getControl(3004).setLabel(info)
        
        self.getControl(3009).setImage(image)
        
        try:
            fanartControl = self.getControl(3005)
            if(fanartControl != None):
                fanartControl.setImage(fanartimage)
        except:
            pass
        
        try:
            thumbControl = self.getControl(3008)
            if(thumbControl != None):
                self.getControl(3008).setImage(thumb)
        except:
            pass
        
        try:
            landscapeControl = self.getControl(3010)
            if(landscapeControl != None):
                self.getControl(3010).setImage(landscapeimage)
        except:
            pass
        
        self.getControl(3006).setImage(clearartimage)
        
        if rating != None:
            self.getControl(3003).setLabel(rating)
        else:
            self.getControl(3003).setVisible(False)
        
        try:
            seasonControl = self.getControl(3015)
            if(seasonControl != None):
                seasonControl.setLabel(str(season))
        except:
            pass
        
        try:
            episodeControl = self.getControl(3016)
            if(episodeControl != None):
                episodeControl.setLabel(str(episodeNum))
        except:
            pass
        
        try:
            resolutionControl = self.getControl(3011)
            if(resolutionControl != None):
                resolution1 =  self.item['streamdetails'].get('video')
                resolution = resolution1[0].get('height')    
                resolutionControl.setLabel(resolution)
        except:
            pass
        
    def setItem(self, item):
        self.item = item
    
    def setCancel(self, cancel):
        self.cancel = cancel
        
    def isCancel(self):
        return self.cancel
            
    def setWatchNow(self, watchnow):
        self.watchnow = watchnow
    
    def isWatchNow(self):
        return self.watchnow
    
    def onFocus(self, controlId):
        pass
        
    def doAction(self):
        pass

    def closeDialog(self):
        self.close()        
        
    def onClick(self, controlID):
        
        xbmc.log("nextup info onclick: "+str(controlID))

        if(controlID == 3012):
            # watch now
            self.setWatchNow(True)
            self.close()
        
        elif(controlID == 3013):
            #cancel
            self.setCancel(True)
            self.close()

        pass
    
    def onAction(self, action):
        
        xbmc.log("nextup info action: "+str(action.getId()))
        if action == ACTION_PLAYER_STOP:
            self.close()
     

